### ISSUE TITLE
security bounty

### ISSUE THREAD

**Issue by tomholub** [FlowCrypt/flowcrypt-docs#26]
- mention that account enumeration is not vulnerability, similar to https://www.drupal.org/node/1004778 (in our case because Public Keys contain emails anyway)

**Comment by seisvelas** [comment]
I agree with this outlook. For example, we are 'susceptible' to the Google dork `site:flowcrypt.com/me` to find all public Encrypted Contact Pages. This is even less of an issue than everyday account discovery, since it's reasonable to assume that someone who publishes an ECP wants people to find and use it. (However, it was somewhat more problematic before we had a CAPTCHA, since it could have facilitated spamming of users via the ECP). 

But I thought perhaps we could use this issue to plan the public bug bounty program that we have been alluding to in the blog (and elsewhere) for some time. For example, would you rather run it independently, or utilize a platform like HackerOne or bugcrowd. 

Let me know your plans for the bug bounty program and I'll start researching / prepping things per your decision.

**Comment by tomholub** [comment]
I would do it as follows:
phrase 1: just a public declaration of bounty rewards and rules (because we don't yet have time to go through many lousy reports we'll get on HackerOne or bugcrowd
phrase 2 lasting maybe 12 months: HackerOne or bugcrowd
phrase 3: back to our own, with updated rules, increased rewards, and inform / invite researchers who have submitted useful bounties over the bounty platform in the past

**Comment by tomholub** [comment]
example issues and rewards I'm thinking:

- an unimportant misconfiguration: $10-$25 (like showing nginx version or something like that)
- a low severity issue $25-$100 (like leaving ssh port open to public unnecessarily)
- a medium severity issue $200-$2000 (like when a bad UX or wording causes user to do something dangerous that they absolutely wouldn't want, possibly account login issues, etc)
- a high severity issue $2000-$5000 (like forging a signature, remotely decrypting like eFail did, exploitable XSS where something private can be retrieved, etc)

**Comment by seisvelas** [comment]
Sounds like a good roadmap. How is this for the initial announcement:

---

## FlowCrypt Bug Bounty Program

FlowCrypt is proud to announce the commencement of our first public bug bounty program! Security bug reports have been an invaluable source for FlowCrypt in the past, and we've always been more than happy to recompense researchers who've taken the time to report security flaws. Perhaps the most significant occurrence was when FlowCrypt user Chris Baker won [$1,000 USD for reporting a browser extension vulnerability](https://flowcrypt.com/blog/article/2019-09-05-browser-extension-vulnerability-report/). We are excited to build on these positive experiences and extend our collaboration with the broader security researcher community. 

For complete, up to date information about scope and rewards, check out the [bug bounty section of our documentation](https://link-to-as-of-yet-imaginary-docs.test). There you can learn about what bugs qualify for a bounty, how to report a vulnerability, and what reward you might expect for your report.

Rewards up to $5,000 USD are available for high severity bugs. We look forward to receiving your reports and working together to strengthen our users' privacy. Any concerns or suggestions related to the program that aren't covered in the documentation can be sent to security@flowcrypt.com

Happy hacking!

\- The FlowCrypt Security Team

---


I intentionally left out specifics regarding money, scope, etc, because that information ought to be in the docs, on a page we keep up to date. If we include those details in the blog, however, people might use that as a reference even if it is outdated. Therefore, to avoid confusion, I push any interested readers to the docs for specifics that might change.


---

## FlowCrypt Public Bug Bounty Program

The FlowCrypt bug bounty program aims to strengthen the privacy protections of our products via open collaboration with security researches. We believe that clear communication, in the form of well defined scope, logically organized vulnerability categories, and fair rewards. With that in mind, we've establish the scope, acceptable vulnerabilities, and rewards below. 
 
#### Scope

**In scope**:

Target | Domain | Source Code
--- | --- | ---
Browser Extension | N/A | flowcrypt-extension
Backend API | flowcrypt.com/api/ | https://github.com/flowcrypt/flowcrypt-backend
Encrypted Contact Page | flowcrypt.com/me/[user] | https://github.com/flowcrypt/flowcrypt-web

**Out of scope**:

* Any Stripe domains
* Actions that affect other user accounts (that you do not control)
* Denial of service against public FlowCrypt services (if you think you have found a DoS vulnerability, contact us and we can help you test it another way)

#### Rewards

****

Thi

---

We want to be mindful of exploits that could involve live customer data, or any financial operations. One common approach is to have a completely sandboxed version of the normal site for researches to hack on. 

This should be really fun to do, although I share some of your ambivalence towards the generally low quality of most reports submitted via platforms like h1 and bugcrowd. For what it's worth, I won $617 via Github's bounty program on HackerOne. If you look at their program rules (https://hackerone.com/github), they describe in great deal the varieties of low severity 'bugs' that don't qualify. As you observe, these kinds of programs receive torrents of inconsequential, low effort submissions. Since we are running our own program, that's of course less of a problem for us (for now).

I link to the repos, including private ones, but I've been wondering: do we plan to eventually open source them once they reach a certain level of maturity. Is that the case? I just put a few random FlowCrypt projects in the scope for examples, we can decide which specific ones actually should be in scope. 

Let me know if you have any feedback for the above and I'll incorporate it!
